### PR TITLE
feat: restrict source request to reputation threshold

### PR DESCRIPTION
**Pull request by capJavert**
AS-134

- restrict `sourceRequest` to reputation threshold
- add new query to check access on frontend (will be used after extension adoption of back compat)

Web integration https://github.com/dailydotdev/apps/pull/2699

TODO

- [x] check if `POST /v1/publications/request` is still needed (confirmed with Ido)

### CODE REVIEW

**Review by omBratteng — APPROVED** [review]

**Review by sshanzel — APPROVED** [review]
Looks good! 🚀

**Review by rebelchris — APPROVED** [review]

**Comment by capJavert** [comment]
I will hold merge until extension adoption.

File: src/errors.ts [hunk 1]
```diff
@@ -63,6 +63,16 @@ export const SubmissionFailErrorMessage: Record<
   ONBOARDING_TAG_LIMIT_REACHED: 'Tag limit reached',
 };
 
+export enum SourceRequestErrorKeys {
+  AccessDenied = 'ACCESS_DENIED',
+}
+
+export const SourceRequestErrorMessage: Record<SourceRequestErrorKeys, string> =
+  {
+    ACCESS_DENIED:
```
Comment (nensidosari) [inline src/errors.ts:72]:
```suggestion
    [SourceRequestErrorKeys.AccessDenied]:
```

**Review by nensidosari — APPROVED** [review]
Looks good, one minor non blocking suggestion

File: src/errors.ts (hunk 1 above)
Comment (capJavert) [reply src/errors.ts]: Added in bf84cac2012c4203be2757fe38f619fe9f9931ad

**Review by idoshamun — APPROVED** [review]

**Pull request merged**